pkg/controllers/provisioning/launcher.go

```diff
@@ -44,11 +45,21 @@ type Launcher struct {
 	CloudProvider cloudprovider.CloudProvider
 }
 
-func (l *Launcher) Launch(ctx context.Context, schedules []*scheduling.Schedule, instanceTypes []cloudprovider.InstanceType) error {
+func (l *Launcher) Launch(ctx context.Context, provisioner *v1alpha5.Provisioner, schedules []*scheduling.Schedule,
+	instanceTypes []cloudprovider.InstanceType) error {
+
 	// Pack and bind pods
 	errs := make([]error, len(schedules))
+	provisioner, err := l.updateState(ctx, provisioner)
+	if err != nil {
+		return fmt.Errorf("unable to determine status of provisioner")
+	}
```

Comment:
I'm not sure why I didn't need this in my earlier revisions. Maybe I was doing something else that was forcing the provisioner to get updated or I wasn't testing my scale ups fast enough, but without this the state in this provisioner object is really off. Since it's hitting a cache, I don't mind moving this into the for loop below either so we reduce the chance of staleness